packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenController.kt

```diff
@@ -0,0 +1,151 @@
+package expo.modules.splashscreen
+
+import android.app.Activity
+import android.os.Handler
+import android.view.View
+import android.view.ViewGroup
+import expo.modules.splashscreen.exceptions.NoContentViewException
+import java.lang.ref.WeakReference
+
+const val SEARCH_FOR_ROOT_VIEW_INTERVAL = 20L
+
+class SplashScreenController(
+  activity: Activity,
+  resizeMode: SplashScreenImageResizeMode,
+  private val rootViewClass: Class<*>,
+  splashScreenConfigurator: SplashScreenConfigurator
+) {
+  private val weakActivity = WeakReference(activity)
+  private val contentView: ViewGroup = activity.findViewById(android.R.id.content) ?: throw NoContentViewException()
```

Comment:
Yeah, there's a little about it here: https://developer.android.com/reference/android/R.id#content
and then we have it referenced here: https://stackoverflow.com/a/4488149
(as I'm traversing the `ContentView`'s hierarchy to find expected RootView.class I don't really care which ContentView is obtained - it could either be DecorView or some of it's children - it doesn't really change anything 😉)